### PR TITLE
Hash incidents fields

### PR DESCRIPTION
**Pull request by yguezpa**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Description
Get incidents and hash some of the fields

### CODE REVIEW

**Comment by kirbles19** [comment]
@erezh31 @yguezpa Done, but seems there's a changelog file missing.

File: Packs/ML/Scripts/HashIncidentsFields/HashIncidentsFields.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    else:
+        entry = res[0]
+        incident_list = json.loads(entry['Contents'])
+        new_incident_list = []
+        for incident in incident_list:
+            for field in fieldsToHash_list:
+                if field in incident:
+                    incident[field] = hash_multiple(incident.get(field))
+            new_ctx = {}
+            ctx = get_context(incident['id'])
```
Comment (erezh31) [inline Packs/ML/Scripts/HashIncidentsFields/HashIncidentsFields.py:47]: please check if context is needed. if `contextKeys` args is empty -  no need to query the context at all

File: Packs/ML/Scripts/HashIncidentsFields/HashIncidentsFields.yml [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+  name: limit
+  required: false
+  secret: false
+- auto: PREDEFINED
+  default: false
+  defaultValue: 'false'
+  description: Whether to query and fetch incident context. Can be "true" or "false".
+    The default is "false".
+  isArray: false
+  name: includeContext
```
Comment (erezh31) [inline Packs/ML/Scripts/HashIncidentsFields/HashIncidentsFields.yml:43]: you don't need this argument, because you query the context separately, remove it

**Comment by yguezpa** [comment]
@kirbles19 Added release note. Can you check it please ?

File: Packs/ML/Scripts/HashIncidentsFields/HashIncidentsFields.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    else:
+        entry = res[0]
+        incident_list = json.loads(entry['Contents'])
+        new_incident_list = []
+        for incident in incident_list:
+            for field in fieldsToHash_list:
+                if field in incident:
+                    incident[field] = hash_multiple(incident.get(field))
+            new_ctx = {}
+            if demisto.args().get('contextKeys', '').split(","):
```
Comment (erezh31) [inline Packs/ML/Scripts/HashIncidentsFields/HashIncidentsFields.py:47]: This condition is not good, the expression always contains empty string.
you need first to extract the context keys like that:
`context_keys = [x for x in demisto.args().get('contextKeys', '').split(",") if x]`
and than you can use `if context_keys: ...`

**Pull request merged**